Find a functionally equivalent code:android.provider.DocumentsProvider.openTypedAssetFile:COMMENT
Method Modifier: public      final       
Comment:/**
 * Implementation is provided by the parent class. Cannot be overriden.
 *
 * @see #openDocumentThumbnail(String, Point, CancellationSignal)
 */

@@ -1,8 +1,9 @@
 {
-    if (opts != null && opts.containsKey(EXTRA_THUMBNAIL_SIZE)) {
-        final Point sizeHint = opts.getParcelable(EXTRA_THUMBNAIL_SIZE);
-        return openDocumentThumbnail(getDocumentId(uri), sizeHint, signal);
+    enforceTree(uri);
+    if (opts != null && opts.containsKey(ContentResolver.EXTRA_SIZE)) {
+        final Point sizeHint = opts.getParcelable(ContentResolver.EXTRA_SIZE);
+        return openDocumentThumbnail(getDocumentId(uri), sizeHint, null);
     } else {
-        return super.openTypedAssetFile(uri, mimeTypeFilter, opts, signal);
+        return super.openTypedAssetFile(uri, mimeTypeFilter, opts);
     }
 }

